pkg/actions/lua/lakefs/catalogexport/table_extractor.lua

```diff
@@ -0,0 +1,186 @@
+local pathlib = require("path")
+local strings = require("strings")
+local yaml = require("encoding/yaml")
+local lakefs = require("lakefs")
+local json = require("encoding/json")
+local common = require("lakefs/catalogexport/common")
+
+-- return partition table from path by based on columns in partition_cols
+function get_partition_values(partition_cols, path)
+    local vals = {}
+    splitted_path = strings.split(path, pathlib.default_separator())
+    for _, part in pairs(splitted_path) do
+        for _, col in ipairs(partition_cols) do
+            local prefix = col .. "="
+            if strings.has_prefix(part, prefix) then
+                vals[col] = part:sub(#prefix + 1)
+            end
+        end
+    end
+    return vals
+end
+
+-- extract partition substr from full path
+function extract_partition_prefix_from_path(partition_cols, path)
+    local partitions = get_partition_values(partition_cols, path)
+    local partitions_list = {}
+    -- iterate partition_cols to maintain order  
+    for _, col in pairs(partition_cols) do
+        table.insert(partitions_list, col .. "=" .. partitions[col])
+    end
+    local partition_key = table.concat(partitions_list, pathlib.default_separator())
+    return partition_key
+end
+
+-- Hive format partition iterator each result set is a collection of files under the same partition
+function lakefs_hive_partition_pager(client, repo_id, commit_id, base_path, page_size, delimiter, partition_cols)
+    local after = ""
+    local has_more = true
+    local prefix = base_path
+    local target_partition = ""
+    return function()
+        if not has_more then
+            return nil
+        end
+        local partition_entries = {}
+        local iter = common.lakefs_object_pager(client, repo_id, commit_id, after, prefix, page_size, delimiter)
+        for entries in iter do
+            for _, entry in ipairs(entries) do
+                is_hidden = pathlib.is_hidden(pathlib.parse(entry.path).base_name)
+                if not is_hidden and entry.path_type == "object" then
```

Comment:
In case of recursive listing, all are objects

Reply:
removed 👍